### PR TITLE
Upgrade to Scala 2.11

### PR DESCRIPTION
**Pull request by stephanh**
Sets all the versions for Scala 2.11. I added some more tests to try and detect problems earlier.

* Things to note are Specs2 depends on scalaz 7.1.1. However scalaz-streams depends on scalaz 7.1.0
* Specs2 depends on scalacheck 1.12.2. However, scalaz-scalacheck-bindings depends on scalacheck 1.11.4
* The scrooge-sbt-plugin for 3.17.0 doesn't seem to be published in the standard places so I am using 3.16.3. I could use 3.18.0 but that seems to depend on an older custom version of libthrift that is also not published in the standard places.
* bijection has a Scala 2.11 version but that does not include bijection-scrooge and after talking to twitter I decided to publish my own.

### CODE REVIEW

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala [hunk 1]
```diff
@@ -15,6 +15,6 @@
 package au.com.cba.omnia.uniform.core.scala
 
 object Scala {
-  val version = "2.10.5"
-  val binaryVersion = "2.10"
+  val version = "2.11.6"
```
Comment (SamRoberts) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala]: align `=` :smile:

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala [hunk 2]
```diff
@@ -28,8 +28,10 @@ object ScalaSettings extends Plugin {
       scalacOptions ++= Seq(
         "-deprecation",
         "-unchecked",
-        "-Ywarn-all",
         "-Xlint",
+        "-Ywarn-dead-code",
```
Comment (SamRoberts) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: What warnings are we discarding here?

File: uniform-thrift/src/sbt-test/thrift/compat/build.sbt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+scala.settings
+
+uniformDependencySettings
+
+strictDependencySettings
+
+uniformThriftSettings
+
+version := "0.1"
+
```
Comment (SamRoberts) [inline uniform-thrift/src/sbt-test/thrift/compat/build.sbt]: unnecessary newlines

File: uniform-dependency/src/sbt-test/dependencies/scalacheck-scalaz/test [hunk 4]
```diff
@@ -0,0 +1 @@
+> test
```
Comment (SamRoberts) [inline uniform-dependency/src/sbt-test/dependencies/scalacheck-scalaz/test]: what is this doing here?

Comment (SamRoberts) [inline uniform-dependency/src/sbt-test/dependencies/scalacheck-scalaz/test]: Oh, ok, this is an sbt command which the scripted framework will automatically run?

**Comment by SamRoberts** [comment]
I would just like to know what warnings we are discarding. If that list seems reasonable I am happy to approve: :+1:

**Comment by vineethvarghese** [comment]
Looks fine to me :+1: except for the issue raised by @SamRoberts . For the dependencies that are not published properly, do we plan to raise an issue against them?

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala (hunk 2 above)
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: Sorry I should have commented that better. There seems to be very little information about it but in Scala 2.11 `-Ywarn-all` was removed. Apparently it is the same as `-Xlint`. I then also added a bunch of extra warnings that aren't part of `-Xlint` and I don't think were part of `-Ywarn-all`.

File: uniform-dependency/src/sbt-test/dependencies/scalacheck-scalaz/test (hunk 4 above)
Comment (stephanh) [inline uniform-dependency/src/sbt-test/dependencies/scalacheck-scalaz/test]: Yep

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala (hunk 1 above)
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala]: Thank you :smile:

**Comment by stephanh** [comment]
I have raised an issue against bijection and they just told me to publish it myself until they will publish their next version.

For scrooge they published a scala 2.11 version for 3.17.0 but then seemed to have reverted back to just Scala 2.10 for 3.18.0.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala (hunk 1 above)
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala]: Fixed.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala (hunk 2 above)
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: Updated the commit comment to include information about this.

File: uniform-thrift/src/sbt-test/thrift/compat/build.sbt (hunk 3 above)
Comment (stephanh) [inline uniform-thrift/src/sbt-test/thrift/compat/build.sbt]: Fixed.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala (hunk 2 above)
Comment (SamRoberts) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: Excellent :+1:

Comment (gavinwhyte) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: Awesome

**Pull request merged**